### ISSUE TITLE
:root { font: 16px/…; } overrides the user’s preferred font size

### ISSUE THREAD

**Issue by fvsch** [csstools/sanitize.css#36]
Just pointing out that the current code overrides the user’s default font-size (that may be set in most browsers’s preferences, including IE, Firefox and Chrome), as would any `font-size` value using units other than `%`, `em`, or `rem` (or `ex` but that’s marginal).

Current Sass code:

``` scss
$root-font-size: 16px !default;
:root {
    font: #{$root-font-size}/#{$root-line-height} $root-font-family;
}
```

Compiled CSS:

``` css
:root{font:1pc/1.5 sans-serif}
```

(I guess the conversion of `16px` to `1pc` is something which minify-css does in order to win just one character. A bit overkill. ^^)

Is this an intentional decision to override user settings?

I understand that not many users will have set up their browser to use a different base text size, and with many websites not respecting this preference they are even less likely to use this preference, but I’d still advise against resetting user settings in boilerplate code.

**Comment by jonathantneal** [comment]
This is a very fair point. Presently, I’ll set the `px` unit on the root to give myself a relative base unit for components, as @chriscoyier does in https://css-tricks.com/rems-ems/. However, this can still be accomplished without explicitly overriding users’ settings.

Would an appropriate change be:

``` scss
$root-font-size: 100% !default;
```

And, for the `20px` override example:

``` scss
$root-font-size: 125%;
```

?

**Comment by tigt** [comment]
That would work perfectly.

**Comment by fvsch** [comment]
Yep, that would be great.

As an aside, in our projects at work we’re using a 20px `rem` base. With sanitize.css we might already set things up like this:

``` scss
$root-font-size: 125%; // So that 1rem=20px with default user settings
@import "path/to/sanitize";

body {
  font-size: .8rem; // 16px equivalent; change for other values
}
```

**Comment by jonathantneal** [comment]
Resolved in 2.1x
